toolchain/check/decl_name_stack.cpp

```diff
@@ -245,6 +260,18 @@ auto DeclNameStack::TryResolveQualifier(NameContext& name_context,
             context_->classes().Get(class_decl->class_id).self_type_id);
         context_->NoteIncompleteClass(class_decl->class_id, builder);
         builder.Emit();
+      } else if (auto interface_decl = context_->insts()
+                                           .Get(name_context.resolved_inst_id)
+                                           .TryAs<SemIR::InterfaceDecl>()) {
+        CARBON_DIAGNOSTIC(
+            QualifiedDeclInUndefinedInterfaceScope, Error,
+            "Cannot declare a member of undefined interface `{0}`.",
+            SemIR::NameId);
+        auto builder = context_->emitter().Build(
+            name_context.parse_node, QualifiedDeclInUndefinedInterfaceScope,
+            context_->interfaces().Get(interface_decl->interface_id).name_id);
```

Comment:
```suggestion
            std::string);
        auto builder = context_->emitter().Build(
            name_context.parse_node, QualifiedDeclInUndefinedInterfaceScope,
            context.sem_ir().StringifyTypeExpr(name_context.resolved_inst_id));
```

Why NameId here but TypeId for the equivalent class diagnostic? Could this be something more like thea bove?

Reply:
Sure. We need to look up the constant value of the interface decl, though, since `StringifyTypeExpr` can't stringify an `InterfaceDecl`.